project-koku/koku
# remove old squashed migrations

When squashing migrations, it is a multistep release process. This PR cleans up old migrations from previous squashes.

When squashing, we are supposed to:
1. create the squashed migration. This consolidates the migrations, and creates a new field in the migration (`replaces`) which tells django which migrations that squashed migration is replacing.
2. We then push that new squashed migration to all of our envs (stage/prod).
3. Next, we delete all the old migrations, and remove the `replaces` from the squashed migration file.
4. Update any new migrations that still point to a replaced migration (e.g. `api 0030_auto_20201007_1403` was still dependent on `0029_auto_20200921_2016` which was replaced in `0001_initial`. The `0030` migration must be updated to depend on `0001` instead of `0029`.)
5. Push these migration changes out to our envs (stage/prod)

This PR accomplished step 3/4 above.

This PR is prep for [COST-2052](https://issues.redhat.com/browse/COST-2052). We should clear out these old migrations before trying to squash anything again.


Testing:
1. `make docker-down`
1. checkout main (`git checkout main`)
2. `make remove-db`
2. `make docker-up-db`
3. `make run-migrations` <- this applies all of the current migrations
4. (optional) create a source and ingest some data
4. checkout this branch (`git checkout remove-old-squashed-migrations`)
5. `make run-migrations` -> nothing should happen:
```
$ make run-migrations    
[standard:public] === Running migrate for schema public
[standard:public] Operations to perform:
[standard:public]   Apply all migrations: api, auth, contenttypes, cost_models, reporting, reporting_common, sessions
[standard:public] Running migrations:
[standard:public]   No migrations to apply.
[standard:acct10001] === Running migrate for schema acct10001
[standard:acct10001] Operations to perform:
[standard:acct10001]   Apply all migrations: api, auth, contenttypes, cost_models, reporting, reporting_common, sessions
[standard:acct10001] Running migrations:
[standard:acct10001]   No migrations to apply.
[standard:template0] === Running migrate for schema template0
[standard:template0] Operations to perform:
[standard:template0]   Apply all migrations: api, auth, contenttypes, cost_models, reporting, reporting_common, sessions
[standard:template0] Running migrations:
[standard:template0]   No migrations to apply.
```
9. `make make-migrations` -> this command should NOT cause any new migrations to be created
```
$ make make-migrations
No changes detected in apps 'api', 'reporting', 'reporting_common', 'cost_models'
```
